apollo-router/src/services/connector_service.rs

```diff
@@ -0,0 +1,322 @@
+        let sf = self
+            .http_service_factory
+            .get(fetch_node.source_id.subgraph_name.as_str())
+            .unwrap()
+            .clone();
+        Box::pin(async move {
+            Ok(process_source_node(sf, fetch_node, connectors, data, current_dir, context).await)
+        })
+    }
+}
+
+pub(crate) async fn process_source_node(
+    http_client: HttpClientServiceFactory,
+    source_node: FetchNode,
+    connectors: Connectors,
+    data: Value,
+    _paths: Path,
+    context: Context,
+) -> (Value, Vec<Error>) {
+    let connector = if let Some(connector) =
+        connectors.get(&SourceId::Connect(source_node.source_id.clone()))
+    {
+        connector
+    } else {
+        return (Default::default(), Default::default());
+    };
+
+    // TODO: remove unwraps
+    let requests = create_requests(connector, &data, context).unwrap();
+
+    let responses = make_requests(
+        http_client,
+        source_node.source_id.subgraph_name.as_str(),
+        requests,
+    )
+    .await
+    .unwrap();
+
+    process_responses(responses, source_node).await
+}
+
+fn create_requests(
+    connector: &Connector,
+    data: &Value,
+    context: Context,
+) -> Result<Vec<HttpRequest>, HttpJsonTransportError> {
+    match &connector.transport {
+        connect::Transport::HttpJson(json_transport) => {
+            Ok(create_request(json_transport, data, context.clone())?)
+        }
+    }
+}
+
+fn create_request(
+    json_transport: &HttpJsonTransport,
+    data: &Value,
+    context: Context,
+) -> Result<Vec<HttpRequest>, HttpJsonTransportError> {
+    Ok(vec![HttpRequest {
+        context,
+        http_request: http_json_transport::make_request(json_transport, data.clone())?,
+    }])
+}
+
+async fn make_requests(
+    http_client: HttpClientServiceFactory,
+    subgraph_name: &str,
+    requests: Vec<HttpRequest>,
+) -> Result<Vec<HttpResponse>, BoxError> {
+    let tasks = requests.into_iter().map(|req| {
+        let client = http_client.create(subgraph_name);
+        async move { client.oneshot(req).await }
+    });
+    futures::future::try_join_all(tasks)
+        .await
+        .map_err(BoxError::from)
+}
+
+async fn process_responses(
+    responses: Vec<HttpResponse>,
+    source_node: FetchNode,
+) -> (Value, Vec<Error>) {
+    let mut data = serde_json_bytes::Map::new();
+    let errors = Vec::new();
+
+    for response in responses {
+        let (parts, body) = response.http_response.into_parts();
+
+        let _ = hyper::body::to_bytes(body).await.map(|body| {
+            if parts.status.is_success() {
+                let _ = serde_json::from_slice(&body).map(|d: Value| {
+                    let (d, _apply_to_error) = source_node.selection.apply_to(&d);
+
+                    // todo: errors
+                    if let Some(d) = d {
+                        // todo: use json_ext to merge stuff
+                        data.insert(source_node.field_response_name.to_string(), d);
```

Comment:
is this inserting different responses into the data using the same key, overwriting the previous response?